Migration to the official GIIS repository of the replication package

augustocristian opened the pull request:
This PR request contains:
- Necessary Java code to execute the experimentation
- In `/docs` folder with:
  - All the generated scenarios as well as their coverage matrix of the user requirements.
  - The test cases generated.
  - The modifications performed in the generated test cases to make them pass.
  - The summary of the results of each RQ
  - The experimental set-up explanation
- The README file that explains briefly the content of the RP
- The citation RF file that enables the "Cite us" button in github with the correct format

augustocristian committed:
Adding the initial content

augustocristian committed:
Removing sonar files, adding the .gitignore and updating the README.md

augustocristian commented:
This is the first version of the RP in the official repository, two doubts that raise to me during the migration:
- Its necessary here sonar-actions?
- May be preferable convert the java multimodule project into a simple java project?
Thanks in advance

javiertuya commented on .gitignore:
```diff
@@ -0,0 +1,209 @@
+# Created by https://www.toptal.com/developers/gitignore/api/intellij,eclipse,java
```
Simplificar gitignore con lo necesario

javiertuya commented on README.md:
```diff
@@ -1,2 +1,125 @@
+The replication package comprises the test scripts used to generate the test scenarios and system test code as well
+as the different inputs required: the user requirements, the system test cases provided as example, the test scenarios
+used as input and the scenario used as example. The replication package also provides the different outputs of our
+exploratory study in the `\docs` folder, the original raw data is available in the [ZENODO](TO-DO) repository.
```
añadir link a zenodo ahora que está

augustocristian replied:
I am waiting for @ClaudiodelaRiva  approval of the Zenodo draft, at the moment it would be available I update everything

javiertuya commented on README.md:
```diff
@@ -1,2 +1,125 @@
+
+## Replication package structure and naming conventions:
+
+The naming conventions are :
```
maybe this should be after the replication package structure

javiertuya commented on README.md:
```diff
@@ -1,2 +1,125 @@
+To the best of our knowledge, FullTeaching has two test suites available in different
+repositories [[1]](https://github.com/elastest/full-teaching) [[2]](https://github.com/codeurjc-students/2019-FullTeaching/tree/Angular-Refactor).
+The test suite used to generate the raw datasets provided in this replication package is a compilation of the available
+test, cases in these repositories. The test suite is made available as the version TO-DO in
```
Use v1.2.0 (I just created it as prerelease)
Use the term release instead of version to match the GitHub terms.

javiertuya commented on README.md:
```diff
@@ -1,2 +1,125 @@
+
+The user requirements are extracted of the FullTeaching documentation (Fuente Pérez, P. (2017). FullTeaching :
+Aplicación Web de docencia con videoconferencia.) and translated to english. The spanish version can be
+consulted [here](/retorch-llm-rp/src/main/resources/input/inputUserRequirements_spa.txt) and the english
+version [here](/retorch-llm-rp/src/main/resources/input/inputUserRequirements_en.txt).
```
Although absolute paths may work, better to use relative paths. Check other places where you also use absolute paths

javiertuya commented on README.md:
```diff
@@ -1,2 +1,125 @@
+
+1. Install java and maven, this experimentation was performed using the following versions:
+    - Maven 3.9.7
+    - Java SE 22.0.1
```
You should not be using non LTS java versions, in this case 21 should preferred

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+    </dependencyManagement>
+    <build>
+        <pluginManagement>
+            <plugins>
```
Do you need these plugins, if not, remove them

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+    </scm>
+
+    <!-- To publish packages (mvn deploy) a profile must be specified (GitHub for snapshots and maven central for releases) -->
+    <profiles>
```
This repo will not generate artifacts for maven, remove these unnecessary parts.

javiertuya commented on retorch-llm-rp/.project:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>samples-giis-template-m1</name>
```
This is still the template name

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>io.github.giis-uniovi</groupId>
+    <artifactId>retorch</artifactId>
+    <version>1.1.1-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <name>retorch</name>
```
Change artifact id and name (e.g. to the name of this repo)

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <modules>
+        <module>retorch-llm-rp</module>
+    </modules>
```
Do you plan to use THIS repo to other works, if so, it is ok to keep it as multimodule, if not, maybe this is not necessary

augustocristian replied:
@javiertuya  I was thinking about rename the current project to retorch-llm-expstudy and leave the name of the repository retorch-llm-rp. What do you think?

javiertuya commented on retorch-llm-rp/pom.xml:
```diff
@@ -0,0 +1,80 @@
+        <name>Software Engineering Research Group (GIIS) - University of Oviedo, ES</name>
+        <url>http://giis.uniovi.es/</url>
+    </organization>
+    <dependencies>
```
Either if this is finally a multimodule or not, check for removal all unnecessary stuff (dependencies and build steps) in the pom (or poms)

javiertuya commented on retorch-llm-rp/pom.xml:
```diff
@@ -0,0 +1,80 @@
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>21</source>
+                    <target>21</target>
```
It seems a little bit inconsistent to use here 21, say that you compile with 22 and set 8 in other places. Just keep java 21 everywhere

javiertuya replied:
Project does not compile

javiertuya commented on retorch-llm-rp/src/test/java/giis/retorch/llmrp/TestClass1.java:
```diff
@@ -0,0 +1,28 @@
+package giis.retorch.llmrp;
```
This should be removed

javiertuya commented on retorch-llm-rp/src/test/resources/log4j2.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Remove if you are not using loggers

augustocristian replied:
I am using loggers in the experimentation classs

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>io.github.giis-uniovi</groupId>
+    <artifactId>retorch</artifactId>
+    <version>1.1.1-SNAPSHOT</version>
```
Change the version number to 1.0.0
After merge, I will create a release with this number (as prerelease) to refer it from zenodo.

augustocristian committed:
Changes requested by reviewer

- .gitignore simplified
- moved naming convenions
- Changed paths to relative
- Removed test class
- Change version to 1.0.0
- Removed profiles not used in pom.xml

augustocristian committed:
Adding the ZENODO repo

augustocristian commented:
I've made the necessary changes and created a new issue (https://github.com/giis-uniovi/retorch-llm-rp/issues/2) to address the restructuring of the project. I treated it as a separate task because doing it now would complicate the review of the current PR.

augustocristian committed:
Minimal mistake in the README.md

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,109 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
```
Project does not compile

augustocristian replied:
I've cloned the repository in my laptop (no previous installation) and both (clean, install with maven work-compile). Also the intellij is able to compile and execute: 
<img width="347" alt="image" src="https://github.com/user-attachments/assets/171050db-fc3e-4be4-93b2-dd3aaf6f0493">
 I am missing something? @javiertuya

augustocristian replied:
I've also executed and works-generate the scenarios: 
<img width="382" alt="image" src="https://github.com/user-attachments/assets/ff402344-0c62-4798-8923-8949f9a5dbb5">

javiertuya replied:
@augustocristian As I have jdk 17, I changed the parent pom version, and it fails. I found that the module pom REDEFINES the java version, this should be removed

augustocristian committed:
Modified both poms

- Restored to the original content-plugins to the m1 and parent pom proyects.
- Removed the plugins not used in the project.

augustocristian commented:
I've compared how the template project is configured and restored  the  initial configuration removing the unnecessary plugins. Sorry for the dependencies update, I've tested and the code remains working without this 3 dependencies updated

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,111 @@
+        </dependencies>
+    </dependencyManagement>
+
+    <build>
```
Still unnecessary dependencies and plugins (both poms)

augustocristian committed:
Removed log4j-slf4j2-impl and maven-javadoc-plugin and thymeleaf

Removed unecesary plugins and dependencies of the project that remained from some early experimentations

augustocristian commented:
Sorry @javiertuya. 
I've removed the unused dependencies from early trials in the project as well as the plugin not used. The  maven-source-plugin is the only that is required to create the jars using maven, that remains in the father pom.xml

javiertuya commented on pom.xml:
```diff
@@ -33,11 +33,6 @@
             </dependency>
-            <dependency>
-                <groupId>org.apache.logging.log4j</groupId>
-                <artifactId>log4j-slf4j2-impl</artifactId>
```
By removing the implementation you are losing the ability to generate logs. Also fix the log settings location

javiertuya commented on pom.xml:
```diff
@@ -0,0 +1,88 @@
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-source-plugin</artifactId>
```
You do not need this to generate a jar that will not be published. Moreover, you do not need to generate a jar

augustocristian replied:
I've completely lost the focus, the pom now is minimized to:
- Only the artifact, group, etc
- The dependencies required
- Our metadata
- Some properties like the maven compiler and target version and the encoding
- License
What can be simplified or removed without impacting the stability of the package manager? @javiertuya

augustocristian replied:
Sorry! I've realized now about the <packaging> annotation, removed and pushed!

augustocristian committed:
Rollback log-4j-imp and removing maven-source plugin

augustocristian commented:
@javiertuya  is something wrong with the last changes? 
I've realized that  there's one unresolved conversation, its not the first time that  I mark something as resolved and appears later as not-resolved.

javiertuya commented:
I marked it as unresolved, you didn't address the second part of the comment

augustocristian committed:
Removing the <packagign> jar anotation

javiertuya commented:
@augustocristian let's talk about this tomorrow during the group meeting

augustocristian committed:
Log location fixed, now under java/resources and working

augustocristian commented:
I've moved the log under /java/resources, now all is working
